### ISSUE TITLE
Outdated Versioning Policy

### ISSUE THREAD

**Issue by iwr-redmond** [Lightning-AI/pytorch-lightning#20670]
### 📚 Documentation

The [Compatibility Matrix](https://lightning.ai/docs/pytorch/stable/versioning.html#compatibility-matrix) on the Versioning Policy documentation page does not mention the 2.5 release series.

The v2.5 release notes [say](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.5.0):

> Lightning 2.5 comes with improvements on several fronts, with **zero** API changes

The v2.5.1 release notes [say](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.5.1):

> bump: testing with latest torch 2.6 (https://github.com/Lightning-AI/pytorch-lightning/pull/20509)

Does this mean that the 2.5 series is compatible with PyTorch 2.6.x, or does the "early visibility" mentioned in the PR refer to upcoming support? Is PyTorch 2.5.x supported?

cc @lantiga @borda

**Comment by iwr-redmond** [comment]
As of #20754, it appears that Pytorch 2.7 is being tested and Pytorch 2.6 is supported. Are we to presume that the versioning policy is about to be replaced?

**Comment by bhimrazy** [comment]
Yes, I believe so — tests already seem to cover PyTorch 2.6+ versions.
@Borda, should we update the versioning docs to reflect this?
> Happy to open a PR.

**Comment by Borda** [comment]
> Does this mean that the 2.5 series is compatible with PyTorch 2.6.x, or does the "early visibility" mentioned in the PR refer to upcoming support? Is PyTorch 2.5.x supported?

yes, Lightning 2.5 is compatible with the latest PT 2.7


> Happy to open a PR.

@bhimrazy, mind send a PR to clarify this

**Comment by speediedan** [comment]
Firstly, thanks to Lightning team for your continued hard work and invaluable contributions to the open-source ecosystem! 

As a follow-up to this issue,  I’ve noticed that recent Lightning minor releases are no longer synchronized with PyTorch’s minor releases as they have been in the past:

| PyTorch Version | Release Date   | Release Artifact                                                                | Lightning Version | Release Date      | Release Artifact                                                                                     | Lag (weeks: Lightning - PyTorch) |
|:---------------:|:--------------|:--------------------------------------------------------------------------------|:-----------------:|:------------------|:-----------------------------------------------------------------------------------------------------|:---------------------------------:|
| 2.0             | Mar 15, 2023  | [PyTorch 2.0.0](https://github.com/pytorch/pytorch/releases/tag/v2.0.0)         | 2.0.0             | Mar 15, 2023      | [Lightning 2.0.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.0.0)              | 0                                |
| 2.1             | Oct 4, 2023   | [PyTorch 2.1.0](https://github.com/pytorch/pytorch/releases/tag/v2.1.0)         | 2.1.0             | Oct 12, 2023      | [Lightning 2.1.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.1.0)              | 1                                |
| 2.2             | Jan 30, 2024  | [PyTorch 2.2.0](https://github.com/pytorch/pytorch/releases/tag/v2.2.0)         | 2.2.0             | Feb 7, 2024       | [Lightning 2.2.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.2.0)              | 1                                |
| 2.3             | Apr 24, 2024  | [PyTorch 2.3.0](https://github.com/pytorch/pytorch/releases/tag/v2.3.0)         | 2.3.0             | Jun 13, 2024      | [Lightning 2.3.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.3.0)              | 7 (50 days)                      |
| 2.4             | Jul 24, 2024  | [PyTorch 2.4.0](https://github.com/pytorch/pytorch/releases/tag/v2.4.0)         | 2.4.0             | Aug 7, 2024       | [Lightning 2.4.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.4.0)              | 2 (14 days)                      |
| 2.5             | Oct 17, 2024  | [PyTorch 2.5.0](https://github.com/pytorch/pytorch/releases/tag/v2.5.0)         | 2.5.0             | Dec 20, 2024      | [Lightning 2.5.0](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.5.0)              | 9 (64 days)                      |
| 2.6             | Jan 29, 2025  | [PyTorch 2.6.0](https://github.com/pytorch/pytorch/releases/tag/v2.6.0)         | —                 | —                 | —                                                                                                    | 5 (40 days)|
| 2.7             | Apr 23, 2025  | [PyTorch 2.7.0](https://github.com/pytorch/pytorch/releases/tag/v2.7.0)         | —                 | —                 | —                                                                                                    | 17 (124 days)|
| 2.8             | Aug 6, 2025*  | [PyTorch 2.8.0-rc3](https://github.com/pytorch/pytorch/tree/v2.8.0-rc3)         | —                 | —                 | —                                                                                                    | 33 (229 days)|


I think both users and maintainers of downstream packages depending on Lightning (I'm the primary maintainer of a downstream Lightning package [Fine-Tuning Scheduler](https://github.com/speediedan/finetuning-scheduler)) would benefit if this apparent versioning policy/cadence change were made explicit in the versioning documentation. 

Can we update the versioning policy to clarify the current minor release policy/cadence for Lightning? E.g., _'minor releases are typically driven by upstream PyTorch breaking changes etc...'_ This would help the community better plan dependencies and upgrades. 
I suspect this will be handled as a separate issue but I'm opening the conversation here first for visibility/relevance. Thanks again for all you do!

@Borda @lantiga @iwr-redmond
